src/vs/editor/browser/services/treeSitter/treeSitterParserService.ts

```diff
@@ -349,7 +358,8 @@ export class TreeSitterTextModelService extends Disposable implements ITreeSitte
 		const environmentService = this._environmentService;
 		await Parser.init({
 			locateFile(_file: string, _folder: string) {
-				return FileAccess.asBrowserUri(`${getModuleLocation(environmentService)}/${FILENAME_TREESITTER_WASM}`).toString(true);
+				const wasmLocation = getWasmLocation(environmentService, FILENAME_TREESITTER_WASM);
+				return FileAccess.asBrowserUri(wasmLocation).toString(true);
```

Comment:
@bpasero, I'm not sure how to make this work on web. On desktop it works as expected (makes a URI that looks like `<path to node modules>/@vscode/tree-sitter-wasm/wasm/tree-sitter.wasm`), but on web, `asBrowserUri` adds some extra duplicate path segments and I end up with `<path to node modules>/@vscode/tree-sitter-wasm/wasm/tree-sitter.js/wasm/tree-sitter.wasm`. Do you have any idea where these extra path segments could be coming from?

Reply:
@jrieken might be able to help

Reply:
Can you take inspiration form [vsda](https://github.com/microsoft/vscode/blob/7473068e0d7b62bfd4ef4c8f6b86832448ff1d36/src/vs/platform/sign/browser/signService.ts#L92) or [textmate](https://github.com/microsoft/vscode/blob/7473068e0d7b62bfd4ef4c8f6b86832448ff1d36/src/vs/workbench/services/textMate/browser/textMateTokenizationFeatureImpl.ts#L375)? They have very explicit branches from web and esm. I believe esp for web a different base-url needs to be assumed.

Reply:
I modeled `getWasmLocation` off of [textmate](https://github.com/microsoft/vscode/blob/7473068e0d7b62bfd4ef4c8f6b86832448ff1d36/src/vs/workbench/services/textMate/browser/textMateTokenizationFeatureImpl.ts#L375). It's supposed to do the same thing as what is done for textmate, but it doesn't do the `FileAccess.asBrowserUri` there as there are places where I think I need to do [`FileAccess.asFileUri`](https://github.com/microsoft/vscode/blob/2430ef04fb1e604c3cf5e91dccf119d4f8a6d612/src/vs/editor/browser/services/treeSitter/treeSitterParserService.ts#L302-L303) instead.

Reply:
The `FileAccess.asBrowserUri` seems to be adding extra path segments, but I can't see where they're coming from.